File: October 2022/Day 26/JAVA Code/523. Continuous Subarray Sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, long k) {
        HashMap<Long,Integer> map = new HashMap<>();
        long sum=0;
        map.put(sum,-1);
        for(int i=0;i<nums.length;i++) {
            long num = nums[i];
            sum = ((sum%k)+(num%k))%k;
            if(map.containsKey(sum)) {
                if(map.get(sum)!=i-1) return true;
            } else {
                map.put(sum,i);
            }
        }
        return false;
    }
}

File: October 2022/Day 26/C++ Code/523. Continuous Subarray Sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> mp;
        int sum =0;
        mp[0]=-1;
        for(int i=0;i<nums.size();i++) {
            sum=((sum%k)+(nums[i]%k))%k;
            if(mp.find(sum)!=mp.end()) {
                if(mp.find(sum)->second!=i-1) return true;
            } else {
                mp[sum]=i;
            }
        }
        return false;
    }
};
